src/Crawler/RepositoryCrawler.cs

```diff
@@ -0,0 +1,208 @@
+﻿using ModelRepoBrowser.Crawler.XmlModels;
+using ModelRepoBrowser.Models;
+using System.Collections.Concurrent;
+
+namespace ModelRepoBrowser.Crawler
+{
+    public class RepositoryCrawler
+    {
+        private static readonly HttpClient client = new();
+        private IDictionary<string, Repository> modelRepositories = new ConcurrentDictionary<string, Repository>();
+
+        /// <summary>
+        /// Parse the Repository tree from the root repository following subsidiary links.
+        /// </summary>
+        /// <param name="rootRepositoryUri">The <see cref="Uri"/> of the repository at the root of the model repository tree.</param>
+        /// <returns></returns>
+        public async Task<IDictionary<string, Repository>> CrawlModelRepositories(Uri rootRepositoryUri)
+        {
+            await CrawlRepositories(rootRepositoryUri, null).ConfigureAwait(false);
```

Comment:
Q: Was geschieht bei nicht erwarteten Exceptions?

Reply:
Das Error handling ist in kommenden Tests noch zu definieren.